src/sgame/sg_admin.cpp

```diff
@@ -2350,6 +2358,92 @@ bool G_admin_setlevel( gentity_t *ent )
 	return true;
 }
 
+bool G_admin_slap( gentity_t *ent )
+{
+	int            pid;
+	double         damage;
+	char           name[ MAX_NAME_LENGTH ], err[ MAX_STRING_CHARS ];
+	gentity_t      *vic;
+	vec3_t         dir;
+
+	if ( level.intermissiontime )
+	{
+		return false;
+	}
+
+	if ( trap_Argc() < 2 )
+	{
+		ADMP( QQ( N_("^3slap:^* usage: slap [^3name|slot#|admin#^7] (^3damage^7)") ) ); 
+		return false; // todo: G_admin_helptext( "slap" ) instead of this copypasta
+	}
+
+	trap_Argv( 1, name, sizeof( name ) );
+
+	if ( ( pid = G_ClientNumberFromString( name, err, sizeof( err ) ) ) == -1 )
+	{
+		ADMP( va( "%s %s %s", QQ( "^3$1$:^* $2t$" ), "slap", Quote( err ) ) );
+		return false;
+	}
+
+	vic = &g_entities[ pid ];
+
+	if ( !admin_higher( ent, vic ) )
+	{
+		ADMP( va( "%s %s", QQ( N_("^3$1$:^* sorry, but your intended victim has a higher admin"
+		          " level than you") ), "slap" ) ); // todo: move this print to a single helper function
+		return false; 
+	}
+
+	if ( G_Team( vic ) == TEAM_NONE || vic->client->pers.classSelection == PCL_NONE )
+	{
+		ADMP( va( "%s %s", QQ( N_("^3$1$:^* cannot slap spectators!") ), "slap" ) );
+		return false; 
+	}
+
+	dir[0] = crandom();
+	dir[1] = crandom();
+	dir[2] = random();
+
+	// todo: G_Knockback
```

Comment:
Either remove the TODO or implement it. Your velocity change seems to be simalr to G_Knockback?